### DISCUSSION TITLE
Console not using the custom dark theme colors

### DISCUSSION THREAD

**Discussion by hnavia** [dbeaver/dbeaver#23351] [Q&A] [answered]
### Description

I've been using DevStyle to install the Dracula Dark Theme (https://draculatheme.com/eclipse) and, until the latest version it was working perfectly but, the latest update, the colors of the text in the console is not working as expected. The following screenshot shows how the colors should look like

![image](https://github.com/dbeaver/dbeaver/assets/69368602/2acf198e-eb8f-4b59-bead-bfb145c91df1)

Here's how the colors actually look like in the console:

![image](https://github.com/dbeaver/dbeaver/assets/69368602/603227cf-4c61-4502-8c33-c7468e6b1a69)

### DBeaver Version

Community Edition 23.1.3.202307231901

### Operating System

Windows 11 with latest updates

### Database and driver

Doesn't apply

### Steps to reproduce

1. Update to the latest DBeaver Community Edition 
2. Select Help, from the top menu, Install New Software
3. In the Work with, search for Darkest Dark Theme - https://www.genuitec.com/updates/devstyle/ci/
4. Select both items: DevStyle Features and Individual Components
5. Install them accepting all the certificates

Although DBeaver uses the color theme, the Console is not showing them. 

Thank you in advance and kind regards.

**Comment by E1izabeth** [comment]
Thank you for bug report.
I can confirm that currently it's reproducible on the latest DBeaver and not reproducible on the latest Eclipse.

**Comment by ayan84** [comment]
Very much irritating for me. Actually hampers my daily work for this theme problem. I used to dark theme but after upgrade devstyle stops working in query editor. Please correct this quickly.

**Comment by hnavia** [comment]
Just to know with no intention to push anyone on this but, when can we expect this issue to be resolved? I don't think it should be to complicated, right?

Thank you in advance and kind regards.

**Comment by hnavia** [comment]
Hi again,

I thought this issue was going to be resolved in version 23.2.1 but the issue still exist:

![image](https://github.com/dbeaver/dbeaver/assets/69368602/1b21fc76-a431-4209-a9a2-6e188d89c53c)

![image](https://github.com/dbeaver/dbeaver/assets/69368602/f24f75ad-2ab8-4467-b46b-add4144c89a0)


Thank you in advance and kind regards.

**Comment by tqllorry** [comment]
me too.The latest version that can support devstyle is 23.0.5

**Comment by wistlo** [comment]
DevStyle is working for me, just only with the Eclipse Standard colors on both Light Gray and Dark Gray.    

I prefer the DevStyle dark style to the native option.

**Comment by hnavia** [comment]
When is this going to be fixed??? It's been several weeks without any updates and I just noticed that this was removed from the 23.3.0 milestone.

**Comment by hnavia** [comment]
Hi, any updates?

**Comment by hnavia** [comment]
Can I get an update please?

**Comment by MashaKorax** [comment]
Hello, @hnavia. Sorry for taking so long to get back to your issue. We plan to fix this in Milestone 23.3.5. Thank you for your patience and attention.

**Comment by hnavia** [comment]
So, it's been removed from the milestone once more...

**Comment by Destrolaric** [comment]
![image](https://github.com/dbeaver/dbeaver/assets/20579256/61f0f956-f16d-47f6-a677-d53fa2eab118)
Hi, I tried to reproduce the mentioned issue. Could you see if `Enable Semantic analysis` is enabled on this page, if not, could you please try to enable it and see if letters are highlighted
![image](https://github.com/dbeaver/dbeaver/assets/20579256/a2d50751-027a-4c4a-8a7e-278f59e35595)

**Comment by hnavia** [comment]
Hi @Destrolaric unfortunately, I can't even install Darkest Dark in the latest DBeaver update 24.0.1. It seems there's an issue with the theme in the repository https://marketplace.eclipse.org/content/darkest-dark-theme-devstyle

I will try as soon as it's fixed. Thanks!

**Comment by vizinq** [comment]
@Destrolaric I'm using v24.0.0 w DevStyle theme. I have Enable Semantic Analytic checked and the DevStyle theme colors are not applied in the console.

**Comment by hnavia** [comment]
I was able to install the DevStyle and Darkest Dark theme. And this is how my configuration looks

![image](https://github.com/dbeaver/dbeaver/assets/69368602/c12a2d9c-3a11-42b9-bfa4-66817375e97d)

The console is still showing odd colors:

![image](https://github.com/dbeaver/dbeaver/assets/69368602/f494ed65-f672-4420-b1c2-250fd12c2b71)

I'm currently using DBeaver 24.0.1

**Comment by E1izabeth** [comment, marked as answer]
The theme you mention seems intended for Eclipse and java code, not for DBeaver. DBeaver doesn't highlight java code - only SQL code.
We have our own colors, which can be changed in preferences `Preferences -> User Interface -> Appearance -> Colors and Fonts`:
![image](https://github.com/dbeaver/dbeaver/assets/28875055/8da8a084-3bb8-46af-85c3-da605434565c)

**Comment by hnavia** [comment]
@E1izabeth then, just for clarification, why is that it worked previously?

On previous version it was working without any problem when using the DevStyle Darkest Dark. And it was even fixed quite a few times. If it's not expected to work I understand but still it's odd that it stopped working at all.

**Reply by E1izabeth** [reply]
> why is that it worked previously?

I have no answer to this question. DevStyle plugin is not an open source, so I can't check their code.

**Reply by hnavia** [reply]
There are so many color options that I'm not sure how to configure the colors appropriately.

This will take a while.
